### PR TITLE
Animate: type getAnimateClassName

### PR DESCRIPTION
**Pull request by sirreal**
## Description

- Rename `useAnimate` to `getAnimateClassName` (publicly `__unstableGetAnimateClassName`) and treat it as a regular function. It's implementation is not like a hook and it doesn't need to be treated as one.
- Type `getAnimateClassName`.

Closes #26965 (supersedes).

## How has this been tested?

- Storybook

## Types of changes

- Internal: Code quality. Type `getAnimateClassName`.
- Internal: Change unstable function name.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/animate/index.js [hunk 1]
```diff
@@ -23,13 +55,15 @@ export function useAnimate( { type, origin = getDefaultOrigin( type ) } ) {
 		);
 	}
 
-	if ( type === 'loading' ) {
-		return classnames( 'components-animate__loading' );
-	}
+	return undefined;
```
Comment (ellatrix) [inline packages/components/src/animate/index.js:58]: What's the use of explicitly returning undefined?

**Review by ellatrix — APPROVED** [review]
Nice!

File: packages/components/src/animate/index.js [hunk 2]
```diff
@@ -3,11 +3,43 @@
...
+ * @typedef {'left' | 'right'} SlideInOrigin
+ * @typedef {{ type: 'appear'; origin?: AppearOrigin }} AppearOptions
+ * @typedef {{ type: 'slide-in'; origin?: SlideInOrigin }} SlideInOptions
+ * @typedef {{ type: 'loading'; origin?: undefined }} LoadingOptions
+ * @typedef {AppearOptions | SlideInOptions | LoadingOptions} UseAnimateOptions
+ */
+
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @param {UseAnimateOptions['type']} type The animation type
```
Comment (ellatrix) [inline packages/components/src/animate/index.js:22]: Looks like this is still for the hook?

File: packages/components/src/animate/index.js [hunk 3]
```diff
@@ -23,13 +55,15 @@ export function useAnimate( { type, origin = getDefaultOrigin( type ) } ) {
...
-	if ( type === 'loading' ) {
-		return classnames( 'components-animate__loading' );
-	}
+	return undefined;
 }
 
 export default function Animate( { type, options = {}, children } ) {
+	deprecated( 'Animate component', {
+		version: '9.6',
+		alternative: 'useAnimate hook',
```
Comment (ellatrix) [inline packages/components/src/animate/index.js:64]: Should still be adjusted.

File: packages/components/src/animate/index.js (hunk 1 above)
Comment (sirreal) [reply packages/components/src/animate/index.js]: Our TypeScript configuration requires a return type when the type is like `string | undefined`. I've changed it to `string | void` which is largely synonymous, but allows the function to omit a return statement and implicitly return `undefined`.

File: packages/components/src/index.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ export {
 export { default as __experimentalAlignmentMatrixControl } from './alignment-matrix-control';
 export {
 	default as Animate,
-	useAnimate as __unstableUseAnimate,
+	getAnimateClassName as __experimentalGetAnimateClassName,
```
Comment (sirreal) [inline packages/components/src/index.js:17]: I just noticed this was `__unstable` - not intended for use outside of Gutenberg. Should it stay that way?

Comment (ellatrix) [reply packages/components/src/index.js:17]: If we're confident in the API, we can stabilise it? The question is, do we intend to use this for future animations, or are there new methods there days? Cc @jasmussen.

Comment (jasmussen) [reply packages/components/src/index.js:17]: I'm sad to say I've not had enough of a chance to leverage this one quite yet, to make a call. However I recall @youknowriad being pretty excited about it?

Comment (sirreal) [reply packages/components/src/index.js:17]: Maybe I can land this PR with the rename and keep it marked unstable. A follow-up could mark it experimental or stabilize.

**Comment by sirreal** [comment]
@ellatrix I know you approved this, but it wasn't ready at the time so I'm disregarding. I can't actually dismiss the review for some reason 🤷‍♂️ 

I do think it's ready for review at this time 🙇‍♂️

File: packages/components/tsconfig.json [hunk 5]
```diff
@@ -11,8 +11,10 @@
 		{ "path": "../primitives" }
 	],
 	"include": [
+		"src/use-animate/**/*",
 		"src/base-control/**/*",
 		"src/dashicon/**/*",
+		// "src/navigation/**/*",
```
Comment (saramarcondes) [inline packages/components/tsconfig.json:17]: Did you mean to include this?

File: packages/components/tsconfig.json [hunk 6]
```diff
@@ -11,8 +11,10 @@
 		{ "path": "../primitives" }
 	],
 	"include": [
+		"src/use-animate/**/*",
```
Comment (saramarcondes) [inline packages/components/tsconfig.json:14]: Should this be just `animate`? I don't see `use-animate`.

Comment (sirreal) [inline packages/components/tsconfig.json:14]:
```suggestion
		"src/animate/**/*",
```

File: packages/components/tsconfig.json (hunk 5 above)
Comment (sirreal) [inline packages/components/tsconfig.json:17]:
```suggestion
```

**Comment by sirreal** [comment]
Thanks  for catching those mistakes, @saramarcondes. I've cleaned them up.

**Comment by ellatrix** [comment]
Consider my approval given for any additional commits. Looks like there's still a few tests failures, but other than that it looks good!

**Comment by sirreal** [comment]
I'll break this into 2 PRs - we shouldn't deprecate `Animate` without stabilizing it's alternative and adding a dev note. I'll focus this PR on typing and renaming the class name helper.

**Pull request merged**